### ISSUE TITLE
Bug with multi server - they start listen port and get connection by another

### ISSUE THREAD

**Issue by webrobot1** [walkor/workerman#865]
Example:

1. Start workerman Websocket server on port 8081 - all ok
![image](https://user-images.githubusercontent.com/20768848/210416619-f0e73a61-7bd9-43f9-bba5-5bf605a09d12.png)

2. Connect to server via another application (not workerman, client on c# language) - all ok
![image](https://user-images.githubusercontent.com/20768848/210416730-f086973b-c2f9-4b37-9cba-d0427252f67d.png)

3. Start another workerman Websocket  server on 8082 - all ok
![image](https://user-images.githubusercontent.com/20768848/210416839-16b5c803-4822-46a6-9418-d5b48eba86ff.png)

4. Server point 1 (8081)  create workerman AsyncTcpConnection to server from pint 3 (with port 8082)  - all ok (doesnt matter wich server connect AsyncTcpConnection  first)
![image](https://user-images.githubusercontent.com/20768848/210416964-f0a17130-c125-4bb5-a93c-476f209fe40a.png)

5. Check after this 8081 port - not ok: 

- Master and Worker of server from point 3 (with 8082 port) start to listen 8081 port too (as Unix console say at the picture)
![image](https://user-images.githubusercontent.com/20768848/210417473-a2e82ffc-9812-4251-b32a-1f9e227eae9e.png)

-  additional this server  from point 3 (port 8082) start to listen my client (what connect only to 8081 server from point 1) (as Lunix console say - you can see t the picture)
![image](https://user-images.githubusercontent.com/20768848/210417726-4c0b3f5b-6758-49a4-8aac-3328240bf705.png)

6. if i stop server from poit 1 (port 8081)  server from point 3 still Listen the port 8081 and client from pint 2 now have Estabilished connection with server 3 (port 8082)
![image](https://user-images.githubusercontent.com/20768848/210418297-1956dfb0-293a-4c94-8b1f-be8aed5aa32c.png)

**Comment by webrobot1** [comment]
Without point 2 situatian is same (server  8082 start to listen 8081 after 8082  get AsyncTcpConnection  from 8081  or  another side of AsyncTcpConnection   will - same result )

**Comment by webrobot1** [comment]
Well the problem is when start two workerman server and some one (doesnt matter who) will open AsyncTcpConnection to another - two servers will start to listen the port of server what start firstly

**Comment by webrobot1** [comment]
As i check it is happen when i try use "exec"  as command to start  one Werkerman from another Workerman

**Comment by webrobot1** [comment]
look like doesnt matter what i try execute from "exec"  command from workerman - new process open and listen the port already (i try exec filу with "Hello World" code and process already in Unix termonal listen the port)

**Comment by webrobot1** [comment]
look like it is problem in php. After "stream_socket_server" (in Worker.php) if you will try "fork" or "exec" or "shell_exec"  command to open new proccess - it will listen the port too . 

But it not open another server - it still one. I was try fork and "unlisten"  it in new process but it unlisen in both process :)

Any idea how start new server from current in new process ?

**Comment by walkor** [comment]
Is the event extension installed?

**Comment by webrobot1** [comment]
> Is the event extension installed?

yes - Swoole. I try open create Timer with "exec"  command to open new process with new workerman but result is same - new process will listen port too after it open (before eny code)

**Comment by walkor** [comment]
Workerman cannot change the features of PHP, such as the process inheritance listening socket when executing exec.
As for the fork problem, some extensions do have such problems, for example, the event extension will affect the behavior of the other processes when you fork process and perform certain operations.

**Comment by webrobot1** [comment]
> Workerman cannot change the features of PHP, such as the process inheritance listening socket when executing exec. As for the fork problem, some extensions do have such problems, for example, the event extension will affect the behavior of the other processes when you fork process and perform certain operations.

i see...Is it possible save connection after Worker reload if it reloadable? Because if PHP work so NEW worker process (what will fork from master)  will have same socket connection as he have before

**Comment by webrobot1** [comment]
or maybe if save connections addres (ip and port)   possible use stream_socket_client  for get resource for new worker

**Comment by webrobot1** [comment]
> PHP can not do that

but php already save the same socket  at multi process, i think it is real...because then i disconnect from one process another still get connection of client (but client not connect exactly to this process, he connect to process what was close)

**Comment by webrobot1** [comment]
![image](https://user-images.githubusercontent.com/20768848/210690749-1c9f0289-6f15-4a1a-946d-6e50a236d9f1.png)
at this pictires - it is two different workerman master +1 worker have same client to connect o same socket

**Comment by webrobot1** [comment]
may be if master process can collect connections too (as worker do)  and after reload worker fork byself with these connections...

**Comment by webrobot1** [comment]
i will create experiment

**Comment by webrobot1** [comment]
> PHP can not do that

well it is 2 way but it possible

1. use https://www.php.net/manual/en/function.socket-wsaprotocol-info-export.php   but it only for windows
2. use https://www.php.net/manual/en/class.ffi.php  and find resource of socket in memory (as you can see it stiil alive at another process)  but need write code in "C"  language

Well for this reason i think i will use Gataway in my project :)

**Comment by webrobot1** [comment]
and normaly open another workerman from current workerman only via http , because as i say open via fork exec system shell - will get port's listening in another process - php work like this
